src/daily_output.ts

```diff
@@ -195,25 +195,23 @@ export class DailyOutput {
 				tokensEarned: 0,
 				tips: 0
 			};
-			// updates the total rewards
-			this.data.ranks.weekly.groupEffortRewards.forEach(({ address, rewards }) => {
-				const prevTotal = this.data.ranks.total.groupEffortRewards.find(
-					({ address: addr }) => addr === address
-				);
-				if (prevTotal) {
-					const indexOfAddress = this.data.ranks.total.groupEffortRewards.indexOf(prevTotal);
-					this.data.ranks.total.groupEffortRewards[indexOfAddress].rewards += rewards;
-				} else {
-					this.data.ranks.total.groupEffortRewards.push({ address, rewards, rankPosition: 0 });
-				}
-			});
-			this.data.ranks.lastWeek = this.data.ranks.weekly;
-			this.data.ranks.weekly = {
-				hasReachedMinimumGroupEffort: false,
-				groupEffortRewards: [],
-				streakRewards: []
-			};
 		}
+		// updates the total rewards
+		this.data.ranks.weekly.groupEffortRewards.forEach(({ address, rewards }) => {
+			const prevTotal = this.data.ranks.total.groupEffortRewards.find(({ address: addr }) => addr === address);
+			if (prevTotal) {
+				const indexOfAddress = this.data.ranks.total.groupEffortRewards.indexOf(prevTotal);
+				this.data.ranks.total.groupEffortRewards[indexOfAddress].rewards += rewards;
+			} else {
+				this.data.ranks.total.groupEffortRewards.push({ address, rewards, rankPosition: 0 });
+			}
+		});
+		this.data.ranks.lastWeek = this.data.ranks.weekly;
+		this.data.ranks.weekly = {
+			hasReachedMinimumGroupEffort: false,
+			groupEffortRewards: [],
+			streakRewards: []
+		};
```

Comment:
Consider breaking these up into private methods like `updateTotalRewardsFromCurrentWeek` and `resetCurrentWeekRanks`.

It may be easier to spot scope errors like this if it reads well

Reply:
5d55a1370987fbb5d79d6f40c487213e39091a32